agent/src/datafile.rs

```diff
@@ -341,17 +374,26 @@ impl DataFile {
         let inner = self.inner.lock().unwrap();
 
         /*
-         * Are we running this snapshot? Fail if so.
+         * Are we running a read-only downstairs for this snapshot? Fail if so.
          */
         if let Some(running_snapshots) =
             inner.running_snapshots.get(&request.id)
         {
-            if running_snapshots.get(&request.name).is_some() {
-                bail!(
-                    "downstairs running for region {} snapshot {}",
-                    request.id.0,
-                    request.name
-                );
+            if let Some(running_snapshot) = running_snapshots.get(&request.name)
+            {
+                match running_snapshot.state {
+                    State::Requested | State::Created | State::Tombstoned => {
+                        bail!(
+                            "read-only downstairs running for region {} snapshot {}",
+                            request.id.0,
+                            request.name
+                        );
+                    }
+
+                    State::Destroyed | State::Failed => {
```

Comment:
Which state here are we looking at?  The state of a downstairs somewhere?

If the state is failed, could it be failed and then come back later and not be failed and then want this snapshot?

Reply:
This is the stored state for a "running snapshot" (aka read only downstairs booted at the snapshot's mount dir: I like and don't like the "running snapshot" name but here we are).

As far as I can tell, there's no path in the `state` state machine to transition out of `Failed` for any of the resources that the agent can interact with.